### PR TITLE
fix(containers/DeleteRessource): fix behaviour when refreshing browser

### PR DESCRIPTION
**Pull request by romainseb**
**What is the problem this PR is trying to solve?**
When we refresh the browser, the desired collection to populate the modal is not always there.
Also, when we refresh the browser, some event needed on saga start are not dispatched
**What is the chosen solution to this problem?**
change function done in render to mapStateToProps to watch the redux change state & re-evaluate the searched resource 
remove the needed event to start the saga. We got the id & the url to redirect with this action catched, so one part is declared in the saga declaration on the sagaRouter, the other part ( id ) is get on routerParams

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**

### CODE REVIEW

File: output/containers.eslint.txt [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+> @talend/react-containers@0.153.0 lint:es /home/travis/build/Talend/ui/packages/containers
 > eslint --config ../../.eslintrc src
 
 The react/require-extension rule is deprecated. Please use the import/extensions rule from eslint-plugin-import instead.
 
 /home/travis/build/Talend/ui/packages/containers/src/ActionSplitDropdown/ActionSplitDropdown.test.js
   4:10  error  'ActionSplitDropdown' is defined but never used  no-unused-vars
 
+/home/travis/build/Talend/ui/packages/containers/src/DeleteResource/DeleteResource.test.js
+  4:15  error  'List' is defined but never used  no-unused-vars
```
Comment (nicomaligne) [inline output/containers.eslint.txt]: OMG !

File: packages/containers/src/DeleteResource/DeleteResource.connect.js [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+	if (ownProps.resourceType) {
+		const resource = state.cmf.collections
+			.get(ownProps.resourceType, new Immutable.Map())
+			.find(currentResource => currentResource.get('id') === ownProps.routeParams.id);
+		return { resource };
+	}
+	return {};
+}
+
+export default cmfConnect({ componentId: 'DeleteResource', mapStateToProps })(Container);
```
Comment (nicomaligne) [inline packages/containers/src/DeleteResource/DeleteResource.connect.js]: componentId: 'DeleteResource' => componentId: ownProps => ownProps.id || ownProps.componentId
Right now the componentId is hardcoded.

File: packages/containers/src/DeleteResource/DeleteResource.container.js [hunk 3]
```diff
@@ -36,15 +36,10 @@ export default class DeleteResource extends React.Component {
...
-			resource = resourcesCollection.find(
-				resoureCollection => resoureCollection.get('id') === id,
-			);
-		}
-		return resource ? { label: resource.get('label'), found: true } : { label: '', found: false };
+	getLabel() {
+		return this.props.resource
+			? { label: this.props.resource.get('label'), found: true }
+			: { label: '', found: false };
 	}
```
Comment (nicomaligne) [inline packages/containers/src/DeleteResource/DeleteResource.container.js]: return {
label: this.props.resource.get('label', '') 
found: !!resource
}
what do u think ?

File: output/containers.eslint.txt (hunk 1 above)
Comment (romainseb) [reply output/containers.eslint.txt]: Oops

File: packages/containers/src/DeleteResource/deleteResource.sagas.js [hunk 4]
```diff
@@ -71,21 +71,21 @@ export function* deleteResourceValidate(uri, resourceType, itemId, resourcePath)
...
+export default function deleteResource({ uri, resourceType, resourcePath, redirectUrl }) {
+	return function* deleteResourceSaga(routerParams) {
 		try {
 			yield race({
 				deleteConfirmationValidate: call(
 					deleteResourceValidate,
 					uri,
 					resourceType,
-					id,
+					routerParams.id,
```
Comment (acateland) [inline packages/containers/src/DeleteResource/deleteResource.sagas.js]: problem i see here is how to handle routes where the the resource indentifier is not strictly called `id`, how to deal with nested routes if we force all resources identifier to be called `id`

example `connection/:id/dataset/:id` with a delete at the end ?

**Comment by acateland** [comment]
couldn't on reload check the existence of the collection inside the sagas ? maybe wait or maybe immediatly redirect ? from the saga

File: packages/containers/src/DeleteResource/deleteResource.sagas.js [hunk 5]
```diff
@@ -71,21 +71,27 @@ export function* deleteResourceValidate(uri, resourceType, itemId, resourcePath)
...
-export default function deleteResource(uri, resourceType, resourcePath) {
-	return function* deleteResourceSaga() {
-		const { redirectUrl, model } = yield take(deleteResourceConst.DIALOG_BOX_DELETE_RESOURCE);
-		const { id } = model;
+export default function deleteResource({
+	uri,
+	resourceType,
+	resourcePath,
+	redirectUrl,
+	routerParamsAttribute,
```
Comment (acateland) [inline packages/containers/src/DeleteResource/deleteResource.sagas.js]: missing in docblock
missing on doc, maybe we should handle a default value ?
or runtime check the parameters ?

**Pull request merged**